cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -62,6 +63,7 @@ type machineController struct {
 	machineSetResource        *schema.GroupVersionResource
 	machineResource           *schema.GroupVersionResource
 	machineDeploymentResource *schema.GroupVersionResource
+	accessLock                *sync.Mutex
```

Comment:
Could make this a non-pointer, has the same effect, plus the zero value works as well so you don't need the initialisation either. I don't mind either way since both work

Reply:
i'll change it in favor a non-pointer reference